src/clause.h

```diff
@@ -20,7 +20,6 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ***********************************************/
 
-
```

Comment:
Hey,

Let me clarify first what we are trying to do: to incorporate the native support for cardinality constraints as begin implemented in MiniCARD into CryptoMiniSat. Mark Liffiton's idea in MiniCARD is that clauses are just special cases of card. constraints, see https://sun.iwu.edu/~mliffito/publications/sat12_liffiton_minicard_poster.pdf
MiniCARD extends the Clause data structure only with the number of watchers.
Fortunately, MiniCARD and CryptoMiniSat are both built on the top of MiniSat's codebase, therefore we do not need to reimplement the CDCL solving part, but only to extend the CDCL-specific methods with a case distinction for card. constraints. Basically, so far we merged those methods from MiniCARD with their counterparts in CryptoMiniSat.

I am afraid that if we created a new data structure for card. constraints in complete separation from clauses then we would need to reimplement certain (all?) CDCL-specific methods only for the new type of constraints. We would like to avoid implementing those since we are not really experts in SAT solving.

Do you surely think that we only need to extend unit propagation? We would also need to extend conflict analysis and clause learning, for instance...

P.S. OK, we are gonna clean the diff, sorry for that.